user-console/src/main/java/org/pentaho/mantle/client/commands/SaveCommand.java

```diff
@@ -83,30 +81,19 @@ protected void performOperationAsync( boolean feedback, final CommandResultCallb
         }
       }
 
-      final FileChooserDialog dialog =
-        new FileChooserDialog(
-          FileChooserMode.SAVE,
-          fileDir,
-          null,
-          false,
-          true,
-          Messages.getString( "save" ),
-          Messages.getString( "save" ),
-          navigatorPerspective.getSolutionTree().isShowHiddenFiles() );
+      final String modeText = isSaveAs ? Messages.getString( "saveAs" ) : Messages.getString( "save" );
+      final boolean showHiddenFiles = navigatorPerspective.getSolutionTree().isShowHiddenFiles();
+
+      final FileChooserDialog dialog = new FileChooserDialog( FileChooserMode.SAVE,
+        fileDir, null, false, true, modeText, modeText, showHiddenFiles );
 
       dialog.setSubmitOnEnter( MantleApplication.submitOnEnter );
 
-      if ( isSaveAs ) {
-        dialog.setTitle( Messages.getString( "saveAs" ) ); //$NON-NLS-1$
-        dialog.setText( Messages.getString( "saveAs" ) ); //$NON-NLS-1$
-      } else {
-        dialog.setTitle( Messages.getString( "save" ) ); //$NON-NLS-1$
-        dialog.setText( Messages.getString( "save" ) ); //$NON-NLS-1$
-      }
+      dialog.setTitle( modeText );
+      dialog.setText( modeText );
```

Comment:
Unfortunately, AFAICT, the code is not equivalent. The OK button label will become "Save As", while, originally, it was always "Save"...
Those two text arguments end up in: `PromptDialogBox( String title, String okText, ...)` (see [here](https://github.com/pentaho/pentaho-commons-gwt-modules/blob/dcefb7876203f7ba748766beed8f625d30c9d2d8/widgets/src/main/java/org/pentaho/gwt/widgets/client/dialogs/PromptDialogBox.java#L42)).

The title argument is passed to `setText`, and so your code preserves behavior w.r.t. to this argument.
However, the `okText` is passed directly to the OK button's text/label.

Later, calling `setTitle` affects some kind of tooltip, which the constructor did not set, initially.
While, calling `setText` call only affects the dialog's "caption"...
The OK button will not be changed again.